### PR TITLE
Nesting locators and adding mouse hover actions

### PR DESCRIPTION
**Pull request by manuelam20**

### CODE REVIEW

File: functions.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// named function
```
Comment (jhenaoz) [inline functions.js]: remove this file

File: src/page/product-list.page.ts [hunk 2]
```diff
@@ -16,7 +18,34 @@ export class ProductListPage{
   }
 
   public async addTshirtToCart(): Promise<ProductAddedModalPage> {
+    const EC = protractor.ExpectedConditions;
+    await browser.wait(EC.elementToBeClickable(this.addToCartButton), 3000);
     await this.addToCartButton.click();
     return new ProductAddedModalPage();
   }
+
+  /*private findByProduct (productName : string) : ElementFinder {
```
Comment (jhenaoz) [inline src/page/product-list.page.ts]: remove this comment

File: test/busy-tshirt.spec.ts [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
     beforeAll(async () => {
       const menuContentPage: MenuContentPage = new MenuContentPage();
       const productListPage: ProductListPage = new ProductListPage();
       const orderSummaryPage: OrderSummaryPage = new OrderSummaryPage();
+      const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
       await menuContentPage.goToTShirtMenu();
-      const productAddedModalPage: ProductAddedModalPage = (await
-      (await productListPage.viewTheStock()).addTshirtToCart());
+      await productListPage.selectProduct('Faded Short Sleeve T-shirts');
+      //  await productListPage.addTshirtToCart();
```
Comment (jhenaoz) [inline test/busy-tshirt.spec.ts]: why this comment?

**Review by jhenaoz — CHANGES_REQUESTED** [review]

File: functions.js (hunk 1 above)
Comment (manuelam20) [reply functions.js]: file deleted

File: src/page/product-list.page.ts (hunk 2 above)
Comment (manuelam20) [reply src/page/product-list.page.ts]: comment deleted

File: test/busy-tshirt.spec.ts (hunk 3 above)
Comment (manuelam20) [reply test/busy-tshirt.spec.ts]: comment deleted

**Review by jhenaoz — APPROVED** [review]

**Pull request merged**